### PR TITLE
/themes - Add standard nav

### PR DESCRIPTION
**Pull request by dkd903**
#### Proposed Changes

* /themes: Integrate the LOHP navigation bar
* 1148-gh-Automattic/martech
* pduY5I-dl-p2

#### Testing Instructions

* Log out
* Navigate to /themes or /themes/free or /theme/archeo
* Ensure that you see the sitewide header Nav bars.
* The page should be unaffected if logged in

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

### CODE REVIEW

File: client/layout/logged-out.jsx [hunk 1]
```diff
@@ -99,6 +99,8 @@ const LayoutLoggedOut = ( {
 		masterbar = null;
 	} else if ( sectionName === 'plugins' ) {
 		masterbar = <UniversalNavbarHeader />;
+	} else if ( sectionName === 'themes' || sectionName === 'theme' ) {
+		masterbar = <UniversalNavbarHeader />;
```
Comment (dkd903) [inline client/layout/logged-out.jsx:106]: Added a separate if section keeping further component props extension in mind.

File: client/layout/universal-navbar-header/nav-style.scss [hunk 2]
```diff
@@ -28,6 +28,21 @@ $studio-blue-30: #0675c4;
 	}
 }
 
+.is-section-themes,
+.is-section-theme {
+	&.has-no-sidebar {
+		.masterbar {
+			height: 56px;
```
Comment (tyxla) [inline client/layout/universal-navbar-header/nav-style.scss:35]: It was unexpected to find that the `nav` element that is a direct child of the masterbar element has a height of 57px:

Screenshot 2022-11-03 at 12 19 45

At the same time, masterbars throughout Calypso are generally 56px indeed. 

Thoughts?

File: client/layout/universal-navbar-header/nav-style.scss [hunk 3]
```diff
@@ -28,6 +28,21 @@ $studio-blue-30: #0675c4;
...
+			position: relative;
```
Comment (tyxla) [inline client/layout/universal-navbar-header/nav-style.scss:36]: In regard to this change, the masterbar in production will remain fixed, together with the search form and filters when the user scrolls down. WIth this change, the masterbar will no longer be fixed, and will be scrolled up. Is that expected?

Comment (tyxla) [inline client/layout/universal-navbar-header/nav-style.scss:36]: Not related to the PR, but I noticed a missing bottom border on the sticky search bar:

Screenshot 2022-11-03 at 12 24 18

Is this a bottle we can pick while we're here?

**Review by tyxla — COMMENTED** [review]
No concerns from the frameworks perspective, just a few questions.

Looks pretty good 👍

File: client/layout/universal-navbar-header/nav-style.scss (hunk 3 above)
Comment (dkd903) [reply client/layout/universal-navbar-header/nav-style.scss:36]: @tyxla - yes, this is similar to the behaviour in logged out /plugins

File: client/layout/universal-navbar-header/nav-style.scss (hunk 2 above)
Comment (dkd903) [reply client/layout/universal-navbar-header/nav-style.scss:35]: Makes sense, added the exception style rule.

**Review by tyxla — COMMENTED** [review]
No further concerns from the framework side 👍 I'll let your team do the final testing and approval if you don't mind.

**Comment by dkd903** [comment]
> No further concerns from the framework side 👍 I'll let your team do the final testing and approval if you don't mind.

Thank you @tyxla

**Review by gmovr — APPROVED** [review]
Changes look good to me. Tested as a logged out and logged in user across various screen sizes both on /themes/?:something and /theme/:something

I did notice that if you go directly to `/themes` as a logged-in user, the logged-out page flashes briefly before you're sent to the site picker. It's not related to this PR, but maybe something we can follow up on. If you go to themes with a site selected (the normal flow most users will go through), this doesn't happen.

**Pull request merged**